can/interfaces/socketcan/socketcan_ctypes.py

```diff
@@ -50,13 +51,56 @@ def __init__(self,
         self.socket = createSocket()
 
         log.debug("Result of createSocket was %d", self.socket)
+        
+        # Add any socket options such as can frame filters
+        if 'can_filters' in kwargs and len(kwargs['can_filters']) > 0:
+            log.debug("Creating a filtered can bus")
+            can_filter_fmt = "={}I".format(2 * len(kwargs['can_filters']))
```

Comment:
I'm tempted to say this should be moved into `socketcan_constants.py` as it is shared with the native interface.